codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/ClockSkewGenerator.java

```diff
@@ -0,0 +1,102 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.*;
+import software.amazon.smithy.go.codegen.integration.*;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.utils.ListUtils;
+
+import java.util.List;
+
+import static software.amazon.smithy.go.codegen.SmithyGoDependency.ATOMIC;
+
+/**
+ * Class to handle clock skew, the discrepancy of time between the client and the server
+ * that can cause SDK calls to fail
+ */
+public class ClockSkewGenerator implements GoIntegration {
+    private static final String TIME_OFFSET = "timeOffset";
+    private static final String ADD_CLOCK_SKEW_BUILD = "addTimeOffsetBuild";
+    private static final String ADD_CLOCK_SKEW_BUILD_MIDDLEWARE = "AddTimeOffsetBuildMiddleware";
+    private static final String ADD_CLOCK_SKEW_DESERIALIZER = "addTimeOffsetDeserializer";
+    private static final String ADD_CLOCK_SKEW_DESERIALIZE_MIDDLEWARE = "AddTimeOffsetDeserializeMiddleware";
+    private static final Symbol TIME_OFFSET_RESOLVER = SymbolUtils.createValueSymbolBuilder(
+            "initializeTimeOffsetResolver").build();
+
+    @Override
+    public void writeAdditionalFiles(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            GoDelegator goDelegator
+    ) {
+
+        ServiceShape service = settings.getService(model);
+
+        // generate code specific to service client
+        goDelegator.useShapeWriter(service, writer -> {
+            generateClockSkewInsertMiddleware(writer);
+            generateClockSkewDeserializeMiddleware(writer);
+            generateTimeOffsetResolver(writer);
+        });
+    }
+
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        ClientMember timeOffset = ClientMember.builder()
+                .name(TIME_OFFSET)
+                .type(ATOMIC.struct("Int64"))
+                .documentation("Difference between the time reported by the server and the client")
+                .build();
+        ClientMemberResolver resolver = ClientMemberResolver.builder()
+                .resolver(TIME_OFFSET_RESOLVER)
+                .build();
+        MiddlewareRegistrar initializeMiddleware = MiddlewareRegistrar.builder()
```

Comment:
We tend to also `static` decl this but this is fine.

Reply:
OK, moved all definitions to be static